src/utils/ar_data_price_regression_estimator.test.ts

```diff
@@ -0,0 +1,78 @@
+import { GatewayOracle } from './gateway_oracle';
+import type { ArweaveOracle } from './arweave_oracle';
+import { expect } from 'chai';
+import { SinonStubbedInstance, stub } from 'sinon';
+import { ARDataPriceRegressionEstimator } from './ar_data_price_regression_estimator';
+import { expectAsyncErrorThrow } from './test_helpers';
+
+describe('ARDataPriceEstimator class', () => {
+	let spyedOracle: SinonStubbedInstance<ArweaveOracle>;
+	let calculator: ARDataPriceRegressionEstimator;
+
+	beforeEach(() => {
+		// Set pricing algo up as x = y (bytes = Winston)
+		spyedOracle = stub(new GatewayOracle());
+		spyedOracle.getWinstonPriceForByteCount.callsFake((input) => Promise.resolve(input));
+		calculator = new ARDataPriceRegressionEstimator(true, spyedOracle);
+	});
+
+	it('can be instantiated without making oracle calls', async () => {
+		const gatewayOracleStub = stub(new GatewayOracle());
+		gatewayOracleStub.getWinstonPriceForByteCount.callsFake(() => Promise.resolve(123));
+		new ARDataPriceRegressionEstimator(true, gatewayOracleStub);
+		expect(gatewayOracleStub.getWinstonPriceForByteCount.notCalled).to.be.true;
+	});
+
+	it('makes 3 oracle calls during routine instantiation', async () => {
+		const gatewayOracleStub = stub(new GatewayOracle());
+		gatewayOracleStub.getWinstonPriceForByteCount.callsFake(() => Promise.resolve(123));
+		new ARDataPriceRegressionEstimator(false, gatewayOracleStub);
+		expect(gatewayOracleStub.getWinstonPriceForByteCount.calledThrice).to.be.true;
+	});
+
+	it('makes three oracle calls after the first price estimation request', async () => {
+		await calculator.getWinstonPriceForByteCount(0);
+		expect(spyedOracle.getWinstonPriceForByteCount.calledThrice).to.be.true;
+	});
+
+	it('getWinstonPriceForByteCount function returns the expected value', async () => {
+		const actualWinstonPriceEstimation = await calculator.getWinstonPriceForByteCount(100);
+		expect(actualWinstonPriceEstimation).to.equal(100);
+	});
+
+	describe('getByteCountForWinston function', () => {
+		it('returns the expected value', async () => {
+			const actualByteCountEstimation = await calculator.getByteCountForWinston(100);
+			expect(actualByteCountEstimation).to.equal(100);
+		});
+
+		it('throws an error when provided winston value is a negative integer', async () => {
```

Comment:
Technically missing a test for fractional winston as well.

Reply:
You can sometimes catch that in the coverage reports by looking at the branch coverage percentage. Unfortunately it doesn't appear to break it down in the branching report by function...